sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/DirectCryptographyClient.cs

```diff
@@ -0,0 +1,489 @@
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual Response<SignResult> Sign(SignatureAlgorithm algorithm, byte[] digest, CancellationToken cancellationToken = default)
+        {
+            var parameters = new KeySignParameters
+            {
+                Algorithm = algorithm.GetName(),
+                Digest = digest
+            };
+
+            return SendRequest(SignOperation, parameters, () => new SignResult() { Algorithm = algorithm }, cancellationToken);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="algorithm"></param>
+        /// <param name="digest"></param>
+        /// <param name="signature"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<VerifyResult>> VerifyAsync(SignatureAlgorithm algorithm, byte[] digest, byte[] signature, CancellationToken cancellationToken = default)
+        {
+            var parameters = new KeyVerifyParameters
+            {
+                Algorithm = algorithm.GetName(),
+                Digest = digest,
+                Signature = signature
+            };
+
+            return await SendRequestAsync(VerifyOperation, parameters, () => new VerifyResult() { Algorithm = algorithm }, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="algorithm"></param>
+        /// <param name="digest"></param>
+        /// <param name="signature"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual Response<VerifyResult> Verify(SignatureAlgorithm algorithm, byte[] digest, byte[] signature, CancellationToken cancellationToken = default)
+        {
+            var parameters = new KeyVerifyParameters
+            {
+                Algorithm = algorithm.GetName(),
+                Digest = digest,
+                Signature = signature
+            };
+
+            return SendRequest(VerifyOperation, parameters, () => new VerifyResult() { Algorithm = algorithm }, cancellationToken);
+        }
+
+        private async Task<Response<TResult>> SendRequestAsync<TContent, TResult>(string operation, TContent content, Func<TResult> resultFactory, CancellationToken cancellationToken)
+            where TContent : IJsonSerializable
+            where TResult : IJsonDeserializable
+        {
+            using DiagnosticScope scope = _pipeline.Diagnostics.CreateScope($"Azure.Security.KeyVault.Keys.Cryptography.RemoteCryptographyClient.{operation}");
+            scope.AddAttribute("key", _keyId.ToString());
+            scope.Start();
+
+            try
+            {
+                using (Request request = CreateRequest(operation))
+                {
+                    request.Content = HttpPipelineRequestContent.Create(content.Serialize());
```

Comment:
it might be worth looking if we can refactor the generic code to a separate method. The code will bloat when JITted,

Reply:
I ended up making the model types classes.  Rereading the code I saw that I was calling extension methods which would have boxed them and eliminated the benefits of them being structs.